src/hooks/use-sorting/use-sorting.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import sortBy from 'lodash/sortBy';
+
+const defaultSortMethod = (items, column) => sortBy(items, column);
+
+const sortItems = (
+  items,
+  column,
+  direction,
+  sortMethod = defaultSortMethod
+) => {
+  if (!column) {
+    return items;
+  }
+
+  const sortedItems = sortMethod(items, column);
+
+  if (direction === 'desc') {
+    return sortedItems.reverse();
+  }
+
+  return sortedItems;
+};
+
+const useSorting = (items, column, sortDirection, sortMethod) => {
+  const initialState = {
+    items: sortItems(items, column, sortDirection, sortMethod),
+    sortedBy: column,
+    sortDirection,
+  };
+
+  const [sortState, setSorting] = React.useState(initialState);
+
+  const onSortChange = columnKey => {
+    let nextSortDirection;
+
+    // if the intented column is not already sorted, the initial direction is 'asc'
+    if (sortState.sortedBy !== columnKey) {
```

Comment:
What if I resort by the same columnKey? Will we recompute it all or should it be a noop?

Reply:
When you click on the same one, it just reverses the `sortDirection`.
I just added a test for this specific case 😇

Reply:
But you're right, in that case can skip the sort (which will still run at least one iteration through all the array) and just do `.reverse()`.